app/src/main/java/com/mongodb/hellosyncrealm/ui/deleteviews/DeleteViewsViewModels.kt

```diff
@@ -24,39 +23,22 @@ class DeleteViewsViewModels(private val realmApp: App) : ViewModel() {
     fun deleteViewCount(count: Int) {
         _isLoading.postValue(true)
 
-        fun onUserSuccess(user: User) {
-            val config = SyncConfiguration.Builder(user, user.id).build()
-
-            Realm.getInstanceAsync(config, object : Realm.Callback() {
-                override fun onSuccess(realm: Realm) {
-                    val visitInfo = realm.where(VisitInfo::class.java).findFirst()
-                    if (visitInfo != null) {
-                        realm.beginTransaction()
-                        visitInfo.apply {
-                            visitCount = if (visitCount.minus(count) >= 0)
-                                visitCount.minus(count)
-                            else
-                                0
-                            _visitInfo.postValue(this)
-                        }
-                        realm.commitTransaction()
-                    }
-                    _isLoading.postValue(false)
-                }
-
-                override fun onError(exception: Throwable) {
-                    super.onError(exception)
-                    //TODO: Implementation pending
-                    _isLoading.postValue(false)
+        viewModelScope.launch(Dispatchers.IO) {
+            val user = realmApp.login(Credentials.anonymous())
+            val config = SyncConfiguration.Builder(user, user.identity).build()
+            val realm = Realm.open(config)
+            val visitInfo = realm.write {
+                val info = query(VisitInfo::class).first().find()!!
```

Comment:
I think you should keep the logic that accounts for the nullable case (i.e avoid the double bang and use `_isLoading.postValue(false)`)